### PR TITLE
Gui dev

### PR DESCRIPTION
**Pull request by DSimonne**
# Pull Request Template

## Description

Summary of the change and which issue it fixes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (will cause existing functionality to not work as expected)

## How has this been tested?
Locally

**Test Configuration**:

* OS:
* Python version:
* ...

## Checklist:

- [ ] I have made corresponding changes to the documentation
- [ ] I have added corresponding unit tests
- [ ] I have run ``doit`` and all tasks have passed

### CODE REVIEW

File: bcdi/postprocessing/facet_analysis.py [hunk 1]
```diff
@@ -1347,8 +1347,8 @@ def to_hdf5(self, path_to_data: str) -> None:
         # Save attributes
         with h5py.File(path_to_data, mode="a") as f:
             try:
-                facets = f.create_group("/data/facets")
-
+                facets = f.create_group("entry_1/process_4/")
```
Comment (carnisj) [inline bcdi/postprocessing/facet_analysis.py:1350]: stupid question, why this choice of path "entry_1/process_4" ?

**Review by carnisj — APPROVED** [review]
I modified few parts in strain.py and preprocess_BCDI.py, in order to avoid KeyError when trying to access a non-existing dictionary key.

File: bcdi/postprocessing/facet_analysis.py (hunk 1 above)
Comment (DSimonne) [reply bcdi/postprocessing/facet_analysis.py:1350]: In the same file there should be an entry_1 (diffraction data); entry_2 (real space data and phase retrieval), and entry_3 postprocessing output (strain)

**Comment by DSimonne** [comment]
I will try a few other backend today to see if I can manage to make it work =)

File: bcdi/postprocessing/facet_analysis.py (hunk 1 above)
Comment (carnisj) [reply bcdi/postprocessing/facet_analysis.py:1350]: Is it written somewhere in the h5 file that this entry corresponds to the facet recognition (something like "entry_1/process_4/name" = "facet_recognition" for example). I am fine with standard naming at the condition that things remain understandable :)

**Review by carnisj — APPROVED** [review]

**Pull request merged**